.github/workflows/build-end-to-end-test.yaml

```diff
@@ -1,51 +1,146 @@
           distribution: 'adopt'
 
-      - name: Run end to end test
+      - uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+
+      - name: Install Allure
+        run: |
+          npm install -g allure-commandline --save-dev
+          allure --version
+
+      - name: Set up Wget
+        if: steps.cache-chromedriver.outputs.cache-hit != 'true' && (success() || failure())
+        run: |
+          sudo apt-get update
+          sudo apt-get install wget
+          wget --version
+
+      - name: Set up Chrome driver
+        if: steps.cache-chromedriver.outputs.cache-hit != 'true' && (success() || failure())
+        run: |
+          wget -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/spotify.gpg
+          echo "deb http://dl.google.com/linux/chrome/deb/ stable main" | sudo tee -a /etc/apt/sources.list.d/google-chrome.list
+          sudo apt-get update -qqy
+          sudo apt-get -qqy install google-chrome-stable
+          CHROME_VERSION=$(google-chrome-stable --version)
+          CHROME_FULL_VERSION=${CHROME_VERSION%%.*}
+          CHROME_MAJOR_VERSION=${CHROME_FULL_VERSION//[!0-9]}
+          sudo rm /etc/apt/sources.list.d/google-chrome.list
+          export CHROMEDRIVER_VERSION=`curl -s https://chromedriver.storage.googleapis.com/LATEST_RELEASE_${CHROME_MAJOR_VERSION%%.*}`
+          curl -L -O "https://chromedriver.storage.googleapis.com/${CHROMEDRIVER_VERSION}/chromedriver_linux64.zip"
+          unzip chromedriver_linux64.zip && chmod +x chromedriver && sudo mv chromedriver /usr/local/bin
+          export CHROMEDRIVER_VERSION=`curl -s https://chromedriver.storage.googleapis.com/LATEST_RELEASE_${CHROME_MAJOR_VERSION%%.*}`
+          curl -L -O "https://chromedriver.storage.googleapis.com/${CHROMEDRIVER_VERSION}/chromedriver_linux64.zip"
+          unzip chromedriver_linux64.zip && chmod +x chromedriver && sudo mv chromedriver /usr/local/bin
+          chromedriver -version
+          which chromedriver
+          which google-chrome
+
+      - name: Generate cache ID
+        run: |
+          echo CACHE_ID=$(date +'%Y-%m-%dT%H:%M:%S') >> $GITHUB_ENV
+
+      - name: Create cache folder
+        continue-on-error: true
+        run: |
+          pwd
+          mkdir ${{ inputs.environment }}-${{ inputs.test }}-test-histories
+          echo "init at - $(date +'%Y-%m-%dT%H:%M:%S')" >> ${{ inputs.environment }}-${{ inputs.test }}-test-histories/init-cache.txt
+
+      - name: Caching test history
+        id: cache-test-history
+        if: always()
+        uses: actions/cache@v3
+        continue-on-error: true
+        with:
+          path: /runner/_work/end-to-end-tests/end-to-end-tests/${{ inputs.environment }}-${{ inputs.test }}-test-histories/
```

Comment:
use `${{ env.GITHUB_WORKSPACE }}` instead of absolute path

https://docs.github.com/en/actions/learn-github-actions/environment-variables